src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -25,6 +60,7 @@ typedef struct MPIDI_XPMEM_seg {
     size_t size;                /* page aligned size */
     void *vaddr;
     MPIDI_XPMEM_segmap_t *segmap_ptr;   /* pointer to my segmap */
+    MPIDI_XPMEM_avl_tree_node_t *node;
```

Comment:
Why don't you directly use `MPIDI_XPMEM_seg_t` as the tree node ?

Reply:
AVL tree node needs extra attributes such as `parent node, left node, right node, min and max segment range`. I don't want to change original data structure too much, so I decided to declare my own node structure.

Reply:
You are adding a pointer access here, which is not cache friendly.

Reply:
I think pointer access is normal for a tree. Is there any better design for this?

Reply:
You can add all tree related attributes in `MPIDI_XPMEM_seg_t`. If you think it makes the code dirty, you can define a macro that declares the additional attributes, and insert the macro in `MPIDI_XPMEM_seg_t`. Apart from additional pointer access, the current code also requires two malloc calls for each segment. It is not good design as this is in performance critical path.

Reply:
OK, as long as you think it is fine to add these structure into `MPIDI_XPMEM_seg_t`, I will change the structure of the current design.

Reply:
Thanks, can you please let me know once all issues are fixed ?